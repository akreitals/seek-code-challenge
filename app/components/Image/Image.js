import PropTypes from 'prop-types';
import Box from '../Box/Box';

const Image = Box.withComponent('img').extend`
    display: block;
    max-width: 100%;
    height: auto;
`;

Image.displayName = 'Image';

Image.propTypes = {
    alt: PropTypes.string.isRequired
};

export default Image;
