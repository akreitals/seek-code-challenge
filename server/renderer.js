import React from 'react';
import ReactDomServer from 'react-dom/server';
import { ServerStyleSheet } from 'styled-components';
import { Provider } from 'react-redux';
import { StaticRouter } from 'react-router-dom';
import { configureStore } from '../app/state/configureStore';
import ThemeProvider from '../app/components/ThemeProvider/ThemeProvider';

import RoutesContainer from '../app/containers/Routes/RoutesContainer';

const isProduction = process.env.NODE_ENV === 'production' || false;

const render = (html, styles) => {
    const assetsManifest =
        process.env.webpackAssets && JSON.parse(process.env.webpackAssets);

    return `
    <!DOCTYPE html>
    <html lang="en">
        <head>
            <meta charset="UTF-8">
            <meta name="viewport" content="width=device-width, initial-scale=1">
            <meta http-equiv="x-ua-compatible" content="ie=edge">
            <title>Seek Coding Challenge</title>
            ${styles}
        </head>
        <body>
            <div id="root">${html}</div>
            <script src='${
                isProduction ? assetsManifest['/vendor.js'] : 'js/vendor.js'
            }'></script>
            <script src='${
                isProduction ? assetsManifest['/app.js'] : 'js/app.js'
            }'></script>
        </body>
    </html>
    `;
};

const serverSideRender = (req, res) => {
    const context = {};
    const sheet = new ServerStyleSheet();
    const store = configureStore();

    const appWithRouter = (
        <Provider store={store}>
            <ThemeProvider>
                <StaticRouter location={req.url} context={context}>
                    <RoutesContainer />
                </StaticRouter>
            </ThemeProvider>
        </Provider>
    );

    if (context.url) {
        return res.redirect(context.url);
    }

    const html = ReactDomServer.renderToString(
        sheet.collectStyles(appWithRouter)
    );

    const styles = sheet.getStyleTags();

    return res.status(200).send(render(html, styles));
};

export default serverSideRender;
