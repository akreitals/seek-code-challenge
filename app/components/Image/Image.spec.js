import React from 'react';
import renderer from 'react-test-renderer';
import Image from './Image';

const imageSrc = 'https://hackclub.com/lah_1.jpg';

describe('Image', () => {
    test('should render a <Image /> component', () => {
        const json = renderer
            .create(<Image src={imageSrc} alt="Alt Text" />)
            .toJSON();
        expect(json).toMatchSnapshot();
    });
});
