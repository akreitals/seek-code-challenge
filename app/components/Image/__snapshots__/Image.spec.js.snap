// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image should render a <Image /> component 1`] = `
<img
  alt="Alt Text"
  className="sc-gzVnrw fQahbn"
  src="https://hackclub.com/lah_1.jpg"
/>
`;
